### DISCUSSION TITLE
Idea: add function to screen facade to get center of screen where mouse cursor is currently located

### DISCUSSION THREAD

**Discussion by danielpetrica** [NativePHP/laravel#373] [Ideas]
Hi, while developing my new project, (a Raycast/Spotlight like app) I found necessar to open a window placed at the center of the screeen currently active. I tried to look into the available facades and didnt't find something perfectly matching my needs so I created thi little (ugly) helper to do it. 

After making it a bit more pretty would it be a good addition to the project? if yes I'd like to contribute that.

```php
<?php

namespace App\Helper;

use Illuminate\Support\Facades\Log;
use Native\Laravel\Facades\Screen;

class ScreenHelper
{
    /**
     * Create a new class instance.
     */
    public static function getMiddleOfActiveScreen(): array
    {

        //get the screen size
        $cursor = Screen::cursorPosition();
        $screen = Screen::displays();

        Log::warning('', ['cursor' => $cursor, 'screen' => $screen]);
        foreach ($screen as $s) {
            /**
             * @var $bouns array
             */
            $bounds = $s['bounds'];
            if (
                $cursor->x >= $bounds['x']
                && $cursor->x <= ($bounds['x'] + $bounds['width'])
                && $cursor->y >= $bounds['y']
                && $cursor->y <= ($bounds['y'] + $bounds['height'])
            ) {
                // I'm inside the display
                return [
                    'x' => $bounds['x'] + $bounds['width'] / 2,
                    'y' => $bounds['y'] + $bounds['height'] / 2,
                ];
            }
        }
        $bounds = $screen[0]['bounds'];

        return [
            'x' => $bounds['x'] + $bounds['width'] / 2,
            'y' => $bounds['y'] + $bounds['height'] / 2,
        ];
    }
}
```

**Comment by simonhamp** [comment]
Don't windows get rendered at the center of the screen already?

**Comment by danielpetrica** [comment]
Yes but I don't think it handles multiple screens. This endpoint returns the center of the screen relative to the screen where the mouse is present. 
I'm not sure what's the default behavior but for me it only opens on my external screen (mac os, I'm not sure about linux) which I think it's my main screen. 
I want to reproduce the raycast behavior of opening on the screen where my mouse is to allow users to have the new window immediately in focus.

**Comment by simonhamp** [comment]
The [PR](https://github.com/NativePHP/laravel/pull/375) for this was merged. Awaiting release
